enhancements/storage/vsphere-driver-configuration.md

```diff
@@ -0,0 +1,269 @@
+by the driver operator. This means that we currently don't allow users to modify the configuration file and need a way to do so.
+
+## Motivation
+
+There is plenty of configuration options that can be set in the configuration file. Some of them are required to be set
+and some of them are optional. In this particular moment we are interested in allowing users to set the maximum limit
+of snapshots that can be created per volume but also consider a more generic approach that would allow users to set
+any configuration option.
+
+Current maximum limit of snapshots per volume is set to 3 by default and if users create 4th snapshot the VolumeSnapshot
+will fail with an error: `the number of snapshots on the source volume XYZ reaches the configured maximum (3)`.
+
+The reason we are interested in a more generic approach is that we expect to receive more requests for allowing users to
+configure other options in the future - even other CSI drivers. Currently, we're aware of the following RFEs:
+
+- AWS EFS CSI usage metrics: https://issues.redhat.com/browse/RFE-3290
+- OpenStack topology: https://issues.redhat.com/browse/RFE-11
+
+### User Stories
+
+- As a cluster administrator, I want to be able to configure the maximum limit of snapshots that can be created per volume
+  so that I can adjust it to my needs.
+- As a cluster administrator, I want to be able to configure any parameter supported by CSI driver.
+
+### Goals
+
+- Allow users to configure the maximum limit of snapshots that can be created per volume for vSphere CSI driver.
+- Allow users to configure any parameter supported by CSI driver.
+- Provide a safe way of configuring any CSI driver while being able to drop any of the parameters in case upstream removes them.
+
+### Non-Goals
+
+- Make things more complicated and easy to break.
+- Allow users to configure feature flags of vSphere CSI Driver.
+- Provide a way to configure drivers in the web UI.
+
+## Proposal
+
+We want to add new fields to ClusterCSIDriver CRD where each field would represent a single
+configuration option. We will basically mirror selected configuration options from the driver into the CRD.
+
+For enabling the feature we will follow API conventions for adding new fields to an existing CRD as described here:
+https://github.com/openshift/enhancements/blob/2894d936eca7c1fcc9cf38e7dc973bbdfa1d88ff/dev-guide/api-conventions.md#new-field-added-to-an-existing-crd
+
+Example:
+
+1. Cluster administrator creates a ClusterCSIDriver object with the maximum limit of snapshots set to 10 using a
+   specific integer field (`clustercsidriver.spec.driverConfig.vSphereConfig.GlobalMaxSnapshotsPerBlockVolume`).
+2. Additionally, they can configure any other option supported by the driver in ClusterCSIDriver - each as a separate
+field that matches driver config ini file values. Administrator does not have to deal with configuring the driver directly.
```

Comment:
I thought we wanted to allow just nr. of snapshots and nothing else.

Reply:
This could be written better to make it clear we intend to allow only 3 options in total for now. I've changed this to make it more clear.